pkg/alertmanager/operator.go

```diff
@@ -654,6 +657,20 @@ func (c *Operator) sync(ctx context.Context, key string) error {
 
 	level.Info(logger).Log("msg", "sync alertmanager")
 
+	// Check if operator has the permission to read the storage class and if AlertManager Specs is using a Volume Claim.
+	if c.canReadStorageClass && am.Spec.Storage.VolumeClaimTemplate.Spec.StorageClassName != nil && *am.Spec.Storage.VolumeClaimTemplate.Spec.StorageClassName != "" {
+		storageClassName := am.Spec.Storage.VolumeClaimTemplate.Spec.StorageClassName
```

Comment:
(nit)
```suggestion
	if storageClassName := operator.StringPtrValOrDefault(am.Spec.Storage.VolumeClaimTemplate.Spec.StorageClassName, ""); c.canReadStorageClass && storageClassName != "" {
```

Reply:
Thanks for suggesting `StringPtrValOrDefault`. Will inculcate this.